monetio/sat/_tempo_l2_no2_mm.py

```diff
@@ -0,0 +1,246 @@
+    """
+    print("reading " + fname)
+
+    ds = xr.Dataset()
+
+    dso = Dataset(fname, "r")
+    lon_var = dso.groups["geolocation"]["longitude"]
+    lat_var = dso.groups["geolocation"]["latitude"]
+    time_var = dso.groups["geolocation"]["time"]
+
+    ds["lon"] = (
+        ("x", "y"),
+        lon_var[:].squeeze(),
+        {"long_name": lon_var.long_name, "units": lon_var.units},
+    )
+    ds["lat"] = (
+        ("x", "y"),
+        lat_var[:].squeeze(),
+        {"long_name": lat_var.long_name, "units": lat_var.units},
+    )
+    ds["time"] = (
+        ("time",),
+        time_var[:].squeeze(),
+        {"long_name": time_var.long_name, "units": time_var.units},
+    )
+    ds = ds.set_coords(["time", "lon", "lat"])
+
+    ds.attrs["reference_time_string"] = dso.time_coverage_start
+    ds.attrs["granule_number"] = dso.granule_num
+    ds.attrs["scan_num"] = dso.scan_num
+
+    if ("pressure" in variable_dict) and "surface_pressure" not in variable_dict:
+        warnings.warn(
+            "Calculating pressure in TEMPO data requires surface_pressure. "
+            + "Adding surface_pressure to output variables"
+        )
+        variable_dict["surface_pressure"] = {}
+
+    for varname in variable_dict:
+        if varname in [
+            "main_data_quality_flag",
+            "vertical_column_troposphere",
+            "vertical_column_stratosphere",
+            "vertical_column_troposphere_uncertainty",
+        ]:
+            values_var = dso.groups["product"][varname]
+        elif varname in [
+            "latitude_bounds",
+            "longitude_bounds",
+            "solar_zenith_angle",
+            "solar_azimuth_angle",
+            "viewing_zenith_angle",
+            "viewing_azimuth_angle",
+            "relative_azimuth_angle",
+        ]:
+            values_var = dso.groups["geolocation"][varname]
+        elif varname in [
+            "vertical_column_total",
+            "vertical_column_total_uncertainty",
+            "fitted_slant_column",
+            "fitted_slant_column_uncertainty",
+            "snow_ice_fraction",
+            "terrain_height",
+            "ground_pixel_quality_flag",
+            "surface_pressure",
+            "tropopause_pressure",
+            "scattering_weights",
+            "gas_profile",
+            "albedo",
+            "temperature_profile",
+            "amf_total",
+            "amf_diagnositc_flag",
+            "eff_cloud_fraction",
+            "amf_cloud_fraction",
+            "amf_cloud_pressure",
+            "amf_troposphere",
+            "amf_stratosphere",
+        ]:
+            values_var = dso.groups["support_data"][varname]
+        elif varname in ["fit_rms_residual", "fit_convergence_flag"]:
+            values_var = dso.groups["qa_statistics"][varname]
+        values = values_var[:].squeeze()
+        fv = values_var.getncattr("_FillValue")
+        if not np.isfinite(fv):
+            values[:][values[:] == fv] = np.nan
```

Comment:
```suggestion
```
From what I've read, this is done automatically by netcdf4-python (though it uses masked array to do it), so no need to redo it here.